tests/acceptance/features/bootstrap/TestingAppContext.php

```diff
@@ -870,6 +875,201 @@ public function theAdministratorHasCreatedFollowingLocks(TableNode $table) {
+	 *
+	 * @return void
+	 */
+	public function theAdministratorAddsUrlAsTrustedServerUsingTheTestingApi($url) {
+		$adminUser = $this->featureContext->getAdminUsername();
+		OcsApiHelper::sendRequest(
+			$this->featureContext->getBaseUrl(),
+			$adminUser,
+			$this->featureContext->getAdminPassword(),
+			'POST',
+			$this->getBaseUrl("/trustedservers"),
+			['url' => $url]
+		);
+	}
+
+	/**
+	 * @Then url :url should be a trusted server
+	 *
+	 * @param string $url
+	 *
+	 * @return  void
+	 */
+	public function urlShouldBeATrustedServer($url) {
+		$trustedServers = $this->getTrustedServers();
+		foreach ($trustedServers as $server => $id) {
+			if ($server === $url) {
+				return;
+			}
+		}
+		\PHPUnit\Framework\Assert::fail("Given url $url is not a trusted server");
+	}
+
+	/**
+	 * @Then the trusted server list should include these urls:
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theTrustedServerListShouldIncludeTheseUrls(TableNode $table) {
+		$trustedServers = $this->getTrustedServers();
+		$expected = $table->getColumnsHash();
+
+		foreach ($expected as $server) {
+			$found = false;
+			foreach ($trustedServers as $url => $id) {
+				if ($url === $server['url']) {
+					$found = true;
+					break;
+				}
+			}
+			if (!$found) {
+				\PHPUnit\Framework\Assert::fail("Given url ${server['url']} is not a trusted server");
+			}
+		}
+	}
+
+	/**
+	 * @Given the administrator has added url :url as trusted server
+	 *
+	 * @param string $url
+	 *
+	 * @return void
+	 */
+	public function theAdministratorHasAddedUrlAsTrustedServer($url) {
+		$adminUser = $this->featureContext->getAdminUsername();
+		OcsApiHelper::sendRequest(
+			$this->featureContext->getBaseUrl(),
+			$adminUser,
+			$this->featureContext->getAdminPassword(),
+			'POST',
+			$this->getBaseUrl("/trustedservers"),
+			['url' => $url]
+		);
+	}
+
+	/**
+	 * @When the administrator deletes url :url from trusted servers using the testing API
+	 *
+	 * @param string $url
+	 *
+	 * @return void
+	 */
+	public function theAdministratorDeletesUrlFromTrustedServersUsingTheTestingApi($url) {
+		$adminUser = $this->featureContext->getAdminUsername();
+		OcsApiHelper::sendRequest(
+			$this->featureContext->getBaseUrl(),
+			$adminUser,
+			$this->featureContext->getAdminPassword(),
+			'DELETE',
+			$this->getBaseUrl("/trustedservers"),
+			['url' => $url]
+		);
+	}
+
+	/**
+	 * @Then url :url should not be a trusted server
+	 *
+	 * @param string $url
+	 *
+	 * @return void
+	 */
+	public function urlShouldNotBeATrustedServer($url) {
+		$trustedServers = $this->getTrustedServers();
+		foreach ($trustedServers as $server => $id) {
+			if ($server === $url) {
+				\PHPUnit\Framework\Assert::fail("Given url $url is a trusted server");
+			}
+		}
+	}
+
+	/**
+	 * @When the administrator deletes all trusted servers using the testing API
+	 *
+	 * @return void
+	 */
+	public function theAdministratorDeletesAllTrustedServersUsingTheTestingApi() {
```

Comment:
I wonder what happens when this is done?
The system-under-test will have no trusted servers defined, so it might block all requests, including from the testing app. That would break the system-under-test and break any later tests.

Reply:
But not having trusted server will not affect the requests from testing app. In general when we run tests the trusted server list is empty, isn't it?

Reply:
My brain is mixed up with `trusted_domains` !
You are corrected - normally trusted servers can be empty and there is no problem.